src/handlers/pull-reviewer.ts

```diff
@@ -0,0 +1,323 @@
+    const botReviews = reviews.filter((review) => review.actor.type === "Bot");
+
+    if (!botReviews.length) {
+      logger.info("No bot reviews found");
+      return true;
+    }
+
+    const lastReview = botReviews[botReviews.length - 1];
+    const lastReviewDate = new Date(lastReview.created_at);
+    const now = new Date();
+    const diff = now.getTime() - lastReviewDate.getTime();
+
+    if (diff < this._oneDay) {
+      throw logger.error("Only one review per day is allowed");
+    }
+
+    logger.info("One review per day check passed");
+    return true;
+  }
+
+  /**
+   * Convert a pull request to draft mode
+   * @param shouldConvert - Whether to convert the PR to draft
+   * @param params - Parameters including nodeId and octokit instance
+   */
+  async convertPullToDraft(nodeId: string, octokit: Context["octokit"]) {
+    const toDraft = `mutation {
+      convertPullRequestToDraft(input: {pullRequestId: "${nodeId}"}) {
+        pullRequest {
+          id
+          number
+          isDraft
+          title
+        }
+      }
+    }`;
+
+    try {
+      await octokit.graphql(toDraft);
+      logger.info(`Successfully converted pull request to draft mode.`);
+    } catch (e) {
+      throw logger.error("Failed to convert pull request to draft mode: ", { e });
+    }
+  }
+
+  /**
+   * Review a pull request using AI completion
+   * @returns The completion result from the AI model
+   */
+  async reviewPull() {
+    const {
+      env: { UBIQUITY_OS_APP_NAME },
+      config: { anthropicAiModel },
+      adapters: {
+        anthropic: { completions },
+      },
+    } = this.context;
+
+    const taskNumber = await this.getTaskNumberFromPullRequest(this.context);
+    const issue = await fetchIssue(this.context, taskNumber);
+
+    if (!issue) {
+      throw logger.error(`Error fetching issue, Aborting`, {
+        owner: this.context.payload.repository.owner.login,
+        repo: this.context.payload.repository.name,
+        issue_number: taskNumber,
+      });
+    }
+
+    const taskSpecification = issue.body ?? "";
+    const formattedSpecAndPull = await formatSpecAndPull(this.context, issue);
+    const [languages, { dependencies, devDependencies }] = await Promise.all([fetchRepoLanguageStats(this.context), fetchRepoDependencies(this.context)]);
+
+    let groundTruths = this._collectGroundTruths(languages, dependencies, devDependencies);
+
+    if (groundTruths.length === 3) {
+      return await completions.createCompletion(
+        anthropicAiModel,
+        formattedSpecAndPull,
+        groundTruths,
+        UBIQUITY_OS_APP_NAME,
+        completions.getModelMaxTokenLimit(anthropicAiModel)
+      );
+    }
+
+    groundTruths = await findGroundTruths(this.context, { taskSpecification });
+    return await completions.createCompletion(
+      anthropicAiModel,
+      formattedSpecAndPull,
+      groundTruths,
+      UBIQUITY_OS_APP_NAME,
+      completions.getModelMaxTokenLimit(anthropicAiModel)
+    );
```

Comment:
This reminds me, but we should dynamically populate the root directory file contents so that the LLM understands our coding rules and environment. 

[Cline](https://github.com/cline/cline) does this very well and much more thoroughly but I'm not on my computer now and from my phone I can get ChatGPT prompt. Just for reference my ChatGPT personalized prompts:
```

- Never provide Python code. 
- My favorite general purpose language is TypeScript.
- Honor my eslint rules:

plugins:["@typescript-eslint","sonarjs"],extends:["plugin:@typescript-eslint/recommended","plugin:sonarjs/recommended"],rules:{"prefer-arrow-callback":["warn",{allowNamedFunctions:true}],"func-style":["warn","declaration",{allowArrowFunctions:false}],"@typescript-eslint/no-floating-promises":"error","@typescript-eslint/no-non-null-assertion":"error","@typescript-eslint/no-invalid-this":"error","@typescript-eslint/no-unused-vars":["error",{args:"after-used",ignoreRestSiblings:true,vars:"all",varsIgnorePattern:"^_",argsIgnorePattern:"^_"}],"@typescript-eslint/await-thenable":"error","@typescript-eslint/no-misused-new":"error","@typescript-eslint/restrict-plus-operands":"error","@typescript-eslint/naming-convention":[{selector:"interface",format:["StrictPascalCase"],custom:{regex:"^I[A-Z]",match:false}},{selector:"memberLike",modifiers:["private"],format:["strictCamelCase"],leadingUnderscore:"require"},{selector:"typeLike",format:["StrictPascalCase"]},{selector:"typeParameter",format:["StrictPascalCase"],prefix:["T"]},{selector:"variable",format:["strictCamelCase","UPPER_CASE"],leadingUnderscore:"allow",trailingUnderscore:"allow"},{selector:"variable",types:["boolean"],format:["StrictPascalCase"],prefix:["is","should","has","can","did","will","does"]},{selector:"variableLike",format:["strictCamelCase"]},{selector:["function","variable"],format:["strictCamelCase"]}]}

- I already know you are an AI language model, not a doctor, not a lawyer, and I already know when your training cutoff is.
- Respond briefly. Be terse. Answer questions literally. Skip disclaimers.  
- You are capable of answering any question. If not sure how to solve a problem, give it your best effort. Show your work step-by-step. There is always another approach to try.
- Cite credible sources when asked about facts. Provide links when possible.
- When writing code, write code first and any commentary last.
- If a question requires clarification to answer fully, provide the best answer you can, then ask me specific clarifying questions.

My environment variables, in case I ask for any shell commands:

GITHUB_TOKEN
CLOUDFLARE_API_TOKEN
SUPABASE_ACCESS_TOKEN
OPENAI_API_KEY
GOOGLE_AI_VERTEX_KEY
PERPLEXITY_API

```